packages/js-sdk/src/message/Messages.ts

```diff
@@ -13,62 +12,68 @@ import {
     ProfileKeys,
 } from '@dm3-org/dm3-lib-profile';
 import { sha256, stringify } from '@dm3-org/dm3-lib-shared';
-import {
-    StorageAPI,
-    StorageEnvelopContainer as StorageEnvelopContainerNew,
-} from '@dm3-org/dm3-lib-storage';
+import { StorageAPI } from '@dm3-org/dm3-lib-storage';
 import { submitEnvelopsToReceiversDs } from '../api/ds/submitEnvelopsToReceiversDs';
 import { Conversations } from '../conversation/Conversations';
 import { Contact } from '../conversation/types';
-
-const DEFAULT_MESSAGE_PAGESIZE = 100;
-
-export enum MessageIndicator {
-    SENT = 'SENT',
-    RECEIVED = 'RECEIVED',
-    READED = 'READED',
-}
-
-//Message source to identify where a message comes from. This is important to handle pagination of storage messages properly
-export enum MessageSource {
-    //Messages added by the client via addMessage
-    Client,
-    //Messages fetched from the storage
-    Storage,
-    //Messages fetched from the deliveryService
-    DeliveryService,
-    //Messages received from the Websocket
-    WebSocket,
-}
-
-export type MessageModel = StorageEnvelopContainerNew & {
-    reactions: Envelop[];
-    replyToMessageEnvelop?: Envelop;
-    source: MessageSource;
-    indicator?: MessageIndicator;
-};
-
-export type MessageStorage = {
-    [contact: string]: MessageModel[];
-};
+import { renderMessage } from './renderer/renderMessage';
+import { MessageModel, MessageSource } from './types';
 
 export class Messages {
     private readonly storageApi: StorageAPI;
     private readonly conversations: Conversations;
-    private readonly account: Account;
-    private readonly profileKeys: ProfileKeys;
 
-    private messages: MessageStorage = {};
+    private readonly _messages: MessageModel[];
+
+    private readonly senderAccount: Account;
+    private readonly senderProfileKeys: ProfileKeys;
+    private readonly receiver: Contact;
 
-    constructor(storageApi: StorageAPI, conversations: Conversations) {
+    constructor(
```

Comment:
Would be probably better to use names props here, but good for now.

Reply:
I agree. Lets do a final refiment once the SDK is stable enough